packages/@uppy/companion/src/server/controllers/s3.js

```diff
@@ -93,13 +96,21 @@ module.exports = function s3 (config) {
       res.status(500).json({ error: 's3: filename returned from `getKey` must be a string' })
       return
     }
+
+    if (typeof config.bucket !== 'function' || typeof config.bucket !== 'string') {
+      res.status(400).json({ error: 's3: bucket key must be a string or a function resolving the bucket string' })
```

Comment:
The error is within the server config, not the client request
```suggestion
      res.status(500).json({ error: 's3: bucket key must be a string or a function resolving the bucket string' })
```